.github/workflows/build.yml

```diff
@@ -0,0 +1,46 @@
+name: PR Builder
+on:
+  pull_request:
+    branches: [ "main" ]
+  push:
+    branches: [ "main" ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    permissions:
```

Comment:
Consider adding `timeout-minutes` to cancel the job as soon as possible if it gets stuck in a loop. Default value is whopping 360 minutes.